Fix #6295: Disable autoplay on Playlist when adding Shared Folder

Brandon-T opened the pull request:
<!-- *Thank you for submitting a pull request, your contributions are greatly appreciated!* -->

## Summary of Changes
- Disable Auto-Play on Playlist when adding Shared Folder
- Cancel loading when the user navigates away from Shared Folder while being added.

<!-- Enter a ticket number for this PR, create a new one if it is not there yet. -->
This pull request fixes #6295

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [x] Light & dark mode
  - [x] Different size classes (iPhone, landscape, iPad)
  - [x] Different dynamic type sizes

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

Brandon-T committed:
Disable autoplay on Playlist and cancel loading when navigating back.

kylehickinson reviewed (changes requested):
The subtasks created should be removed in favour of `async let`

Brandon-T committed:
Fix cancellation, so that `self` can `deinit` properly the moment the task is cancelled, and all operations stop immediately.
This is because the `Task` is capturing `self` and keeping it alive for the duration of the task which pauses cleanup and causes audio glitches.
